hil/client/switch.py

```diff
@@ -23,7 +67,65 @@ def register(self, switch, subtype, *args):
 #       of the switches HIL will control and has read the
 #       HIL documentation to use appropriate flags to register
 #       it with HIL.
-        raise NotImplementedError
+        switch_api = "http://schema.massopencloud.org/haas/v0/switches/" + subtype
```

Comment:
@zenhack @naved001 my first question is here, as Ian said a while ago, a user who is using the client library should type, for example, `NEXUS`. However, our `cli` is directly calling the client library (based on how to structure HIL) and I don' think a user who is using the `cli` would type `NEXUS` (constant that contains the switch api uri) My question is how I should proceed and is the current approach that works?

Reply:
the user currently types `nexus` (lowercase), we could just take the user input and do a `input.upper` before the checks/comparisons.

Reply:
right, but the `NEXUS` in the client library contains the full uri, including the `schema.massopen.cloud`

Reply:
okay, got you! Thanks!

Reply:
@naved001 quick question, how do you match a string with a variable name.

For example,

we have `"nexus"`.uppercase() = `"NEXUS"` as a uppercase case string. And we have a global variable called `NEXUX` that equals to "http://schema.massopencloud.org/haas/v0/switches/nexus".

There is no mapping between them right?

Reply:
oh sorry for the confusion. The user types in the string "nexus", we append it to the url and then compare it against the value of the variable name, does that make sense?